### PR TITLE
Website: dark mode #127

### PR DESCRIPTION
**Pull request by LeeWannacott**
I basically went with a solarized dark theme like in GNOME Terminal. The apple logo is coloured(colored for american's) white by using filter to invert the colour; probably not the best implementation. I might write a script that can change fill from SVG files sounds like a good little project...

Let me know if you want changes. Cheers.

### CODE REVIEW

**Comment by strager** [comment]
Style opinions about dark mode:
* The background colors for the feature table and the installation comparison table are too bright. Darken them.
* Visited link color is low contrast. Make visited links brighter. Maybe also lighten unvisited links too.
* The demo page doesn't have dark mode. :cry:

File: website/index.html [hunk 1]
```diff
@@ -112,6 +112,17 @@
...
+
+    @media (prefers-color-scheme: dark) {
+      a:link {color:#268BD2;}
+      a:visited {color:#6C71C4;}
+      body {background-color:  #002B36; color: white;}
+      td {color:black;}
+    }
+    @media (prefers-color-scheme: light) {
+      body {background:#fff; color: #333; }
+    }
```
Comment (strager) [inline website/index.html:124]: Could we move most of this into main.css?

Comment (strager) [inline website/index.html:124]: Nit: Format by running `yarn fmt`.

File: website/install.html [hunk 2]
```diff
@@ -100,6 +100,19 @@
       .os-column-table > .windows {
         grid-area: windows;
       }
+
+    @media (prefers-color-scheme: dark) {
+      a:link {color:#268BD2;}
+      a:visited {color:#6C71C4;}
+      body {background-color:  #002B36; color: white;}
+      td {color:black;}
+      .macos-logo-color {filter: invert(85%);}
```
Comment (strager) [inline website/install.html:109]: Nit: Can we color-match the content logo from apple.com? https://www.apple.com/v/home/s/images/logos/watch-series-6/logo_hero__trsrpeu6koqy_medium_2x.png

File: website/install.html [hunk 3]
```diff
@@ -1019,7 +1032,7 @@ <h2>
             alt="GNU/Linux"
             title="GNU/Linux"
           />
-          <img class="os-logo" src="macos.svg" alt="macOS" title="macOS" />
+          <img class="os-logo macos-logo-color" src="macos.svg" alt="macOS" title="macOS" />
```
Comment (strager) [inline website/install.html:1035]: Nit: I'd call this class just `macos-logo`. The fact that it changes the color is an implementation detail.

File: website/install.html (hunk 2 above)
Comment (strager) [inline website/install.html:109]: Instead of creating a class, could we select based on the `src` attribute? For example, `.os-logo[src="macos.svg"]`

Comment (LeeWannacott) [reply website/install.html]: I color matched (it is pretty close to white) by figuring out the color of the logo which is #fafafa and then changing the svg file to the opposite hex value which is pretty close to black and then inverting that to get #fafafa.

File: website/index.html (hunk 1 above)
Comment (LeeWannacott) [reply website/index.html]: I moved most of it into main  darkening table color's didn't seem to work when they were in main though.

File: website/install.html (hunk 2 above)
Comment (LeeWannacott) [reply website/install.html]: I ended up doing this therefore not needing class macos-logo.

**Comment by strager** [comment]
Blocking: demo/index.html: In dark mode, error highlights look bad.

Blocking: demo/index.html: In light mode, code is light gray and impossible to read.

Otherwise, looks good!

File: website/demo/index.html [hunk 4]
```diff
@@ -55,6 +55,24 @@
...
+        text-decoration-style: wavy;
+      }
+      mark:empty::before {
+        color: red;
+        content: url("carat.svg") / "^";
+        content: url("carat.svg");
+        margin-left: -2px; /* (width of SVG) / 2 */
+        margin-top: 4px; /* height of SVG */
+        position: absolute;
+      }
```
Comment (strager) [inline website/demo/index.html:74]: Blocking: Remove duplicated rules.

File: website/demo/index.html [hunk 5]
```diff
@@ -82,6 +99,24 @@
...
+        a:visited {
+          color: #979ef8;
+        }
+        #code-input {
+          color: white;
+        }
+        body {
+          background-color: #002b36;
+          color: white;
+        }
```
Comment (strager) [inline website/demo/index.html:115]: Isn't this already in main.css? Why duplicate the rules?

File: website/license.html [hunk 6]
```diff
@@ -1,5 +1,25 @@
...
+        background-color: #002b36;
+        color: white;
+      }
+    }
+    @media (prefers-color-scheme: light) {
+      body {
+        background: #fff;
+        color: #333;
+      }
+    }
```
Comment (strager) [inline website/license.html:21]: Isn't this already in main.css? Why duplicate the rules?

File: website/main.css [hunk 7]
```diff
@@ -75,3 +75,26 @@ main > h2:first-child {
...
+  }
+  .os-logo[src="macos.svg"] {
+    filter: invert(100%);
+  }
+}
+
+@media (prefers-color-scheme: light) {
+  body {
+    background: #fff;
+    color: #333;
```
Comment (strager) [inline website/main.css:98]: Minor: Why are we changing the text color of light mode? Making it explicitly black is fine, but you're making it off-black here.

File: website/main.css [hunk 8]
```diff
@@ -75,3 +75,26 @@ main > h2:first-child {
...
+  a:visited {
+    color: #979ef8;
+  }
+  body {
+    background-color: #002b36;
+    color: white;
+  }
+  .os-logo[src="macos.svg"] {
+    filter: invert(100%);
+  }
```
Comment (strager) [inline website/main.css:92]: Fancy! :+1:

**Review by strager — CHANGES_REQUESTED** [review]

File: website/main.css (hunk 8 above)
Comment (LeeWannacott) [reply website/main.css:92]: Thanks for teaching how to use source. The filter invert is a bit of a hack : ].

File: website/main.css (hunk 7 above)
Comment (LeeWannacott) [reply website/main.css]: Have made text explicitly black and background explicitly white. Have left prefers-color -scheme light in-case you want to change colors later on.

File: website/license.html (hunk 6 above)
Comment (LeeWannacott) [reply website/license.html]: Duplicated rules have been deleted. I left <style></style> tags just in-case you end up changing style later on. However, probably unnecessary.

File: website/demo/index.html (hunk 5 above)
Comment (LeeWannacott) [reply website/demo/index.html]: Duplicated rules have been deleted.

File: website/demo/index.html (hunk 4 above)
Comment (LeeWannacott) [reply website/demo/index.html]: Duplicated rules have been deleted.

**Comment by strager** [comment]
Merged as commit 8827a9868.

**Pull request closed without merging**